.github/workflows/android-check.yaml

```diff
@@ -46,6 +46,13 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 1
+          sparse-checkout: "3party\nandroid\nbase\ncmake\ncoding\ncppjansson\ndata\ndescriptions\n\
```

Comment:
I don't think that optimizing full builds have many benefits, it would be hard to support it. A better solution could be to use ccache.